include/glow/Runtime/DeferredWeightLoader.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_DEFERREDWEIGHTLOADER_H
+#define GLOW_RUNTIME_DEFERREDWEIGHTLOADER_H
+
+#include "glog/logging.h"
+
+#include "glow/Base/Tensor.h"
+#include "glow/Support/Error.h"
+#include "glow/Support/Register.h"
+#include "glow/Support/Support.h"
+
+namespace glow {
+namespace runtime {
+
+/// A base class for deferred weight loaders. This allows for large weights to
+/// be skipped during compilation and loaded after compilation one at a time.
+class DeferredWeightLoader {
+public:
+  /// Loads the next weight, returns an Error indicating success/failure. Frees
+  /// any resources used by the current deferred weight.
+  virtual Error loadNextWeight();
```

Comment:
I kind of think these methods should all be pure virtual instead of having implementations that do nothing but throw or return dummy values.  You'd never actually want to instantiate the base class.